tests/profiler/bob_user_workflow_verbose_profiler_config.yml

```diff
@@ -0,0 +1,216 @@
+# Note the call to Profiler.profile() takes multiple batches
+# Unless filtered / specified this list of batches will be used for all metrics/parameters
+# This profiler is meant to be used on the NYC taxi data (yellow_trip_data_sample_2019-*.csv)
+variables:
+  max_amount: 99999.99
+  false_positive_threshold: 0.01
+# TODO: Support lists and mappings in variables
+  quantiles:
+    - 0.05
+    - 0.25
+    - 0.50
+    - 0.75
+    - 0.95
+rules:
+  # Here we define 3 rules showing various methods to handle multi-batch parameters
+  # 1. my_rule_for_ids uses a simple ParameterBuilder to build a list of parameters per batch to be aggregated in the referencing ExpectationConfigurationBuilder
+  # 2. my_rule_for_amounts uses an aggregating ParameterBuilder to build a list of parameters and then aggregate them using a specified aggregation function. (While this example is not sufficiently complex to warrant a custom ParameterBuilder, one could be created - alternatively the complexity could be handled in a custom ExpectationConfigurationBuilder)
+  # 3. my_rule_for_numeric_data uses MultiBatchBootstrappedMetricDistributionParameterBuilder to compute multiple parameters to supply ExpectationConfigurationBuilders with pre-computed parameters
+
+  my_rule_for_ids:
+    # Create a value_set based on prior batches (relative to when the profiler is run) and evaluate new batches using this value_set
+    # Rationale: I want alerts when new value_set items are added to my dataset
+    class_name: Rule
+    module_name: great_expectations.profiler.rule.rule
+    domain_builder:
+      # Build list of domains from some categorical column domains: vendor_id, rate_code_id, pickup_location_id, dropoff_location_id, payment_type
+      class_name: SimpleColumnSuffixDomainBuilder
+      module_name: great_expectations.profiler.domain_builder.simple_column_suffix_domain_builder
+      column_name_suffixes:
+        - _id
+        - _type
+    parameter_builders:
+      - name: my_value_set_for_ids
+        # This parameter is the union of all observed batch value_sets
+        # MultiBatchParameterBuilder does not aggregate Parameters, it returns a list of value_set for each batch
+        class_name: MultiBatchParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_parameter_builder
+        metric_domain_kwargs: $domain.domain_kwargs
```

Comment:
Ah - this just clicked for me. We can pass anything we want to metric_domain_kwargs, and thus get metrics for domains not built by the domain builder

Reply:
OH GREAT POINT! That makes total sense. I'm not sure if that example goes into Alice / Bob or another more advanced User Workflow but we should add that config soon.

Reply:
I guess this could also work for the "I want to pin this parameter to a specific domain within the domains built by the domain builder"

Reply:
@anthonyburdi @talagluck -- I think that we are saying "While domains are built only by the configured domain builder, the parameters can rely on any metric, specified by `metric_name` and `metric_domain_kwargs`." -- is this correct?

Reply:
Well said @alexsherstinsky! Yes I think that is correct and we should put this note in documentation.